lib/mirage.ml

```diff
@@ -729,13 +729,13 @@ let ipv4_qubes db ethernet arp = ipv4_qubes_conf $ db $ ethernet $ arp
 
 let ipv6_conf ?addresses ?netmasks ?gateways () = impl @@ object
     inherit base_configurable
-    method ty = ethernet @-> time @-> mclock @-> ipv6
+    method ty = ethernet @-> random @-> time @-> mclock @-> ipv6
     method name = Name.create "ipv6" ~prefix:"ipv6"
     method module_name = "Ipv6.Make"
     method packages = Key.pure [ package ~sublibs:["ipv6"] "tcpip" ]
```

Comment:
Re `1`: You're right (and my thinking was too muddy on this), but I think we can help the situation by introducing a conflict -- mirage 3.0.0 conflicts with tcpip > 3.0.0 .  This still isn't quite right (as you say, the dependency is actually in the generated unikernel, but it's too late to do anything about that), but it should avoid the situation where a user has installs `mirage` and generates a unikernel that the `tcpip` they already have will be unable to use.

Your point re: `3` is well-taken, thanks for that.